src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion,) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
 
-class DeliveryDrone {}
+    // this.coords = {
+    //   x,
+    //   y,
+    // };
+
+    this.chipVersion = chipVersion;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name},`
+      + ` Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords,
+    chipVersion, maxLoadWeight, currentLoad = null) {
```

Comment:
fix your params formatting
```suggestion
  constructor(
    name, 
    weight, 
    coords,
    chipVersion, 
    maxLoadWeight, 
    currentLoad = null
  ) {
```